src/libsyntax_pos/symbol.rs

```diff
@@ -814,12 +814,6 @@ impl Decodable for Symbol {
     }
 }
 
-impl<T: std::ops::Deref<Target=str>> PartialEq<T> for Symbol {
```

Comment:
When this is merged, there's some possibility of angry @Manishearth appearing and starting complaining about clippy-specific symbols.

We should probably do this anyway and see, perhaps it doesn't introduce too much inconvenience or clippy-specific helpers can reduce it.

Reply:
I don't think we have those. But please test the clippy build here before merging

Reply:
@Manishearth How do I test a clippy build? I tried doing `cargo build` in `tools/clippy/` but got bazillions of compile errors, even without this PRs commits applied.

Reply:
Should be `x.py build src/tools/clippy`, may not work in stage 1 (not sure).

Reply:
Also, to be clear, it's fine if clippy breaks: Just make sure it's fixable :smile: . (It's fine if it doesn't get fixed in this PR itself, too, as long as clippy isn't using Symbol in a way incompatible with the new stuff)

Reply:
Clippy build will not work in stage 0.

It'd be total overkill in this case but if you face issues and really want to test it there are Docker images (`./src/ci/docker/run.sh x86_64-gnu-tools`) but it will take long time on slow machines.

Reply:
I took a look. There are lots of compile errors but I'm confident clippy can be unbroken relatively easily.

The most common case is `name = "foo"` comparisons, where `name` is a `Symbol`. If `"foo"` has a static symbol in the compiler, it can be changed to `name = sym::foo`. Otherwise, it can be changed to `name.as_str() = "foo"`.

The other common case is `check_name("foo")`, which can become `check_name(sym::foo)` if there is a static symbol in the compiler, or `check_name(Symbol::intern("foo"))` otherwise.

Alternatively, clippy could predefine its own symbols and use those.